File: 3_3/p020.py
# def main(n, l):
#     cnt = 0
    # for a in range(n):
    #     for b in range(a+1, n):
    #         for c in range(b+1, n):
    #             for d in range(c+1, n):
    #                 for e in range(d+1, n):
    #                     if l[a]+l[b]+l[c]+l[d]+l[e] == 1000:
    #                         cnt += 1

# if __name__ == '__main__':
#     n = int(input())
#     l = list(map(int, input().split()))
#     main(n, l)
N = int(input())
A = list(map(int, input().split()))
 
dp = [[0] * 1001 for _ in range(6)]
dp[0][0] = 1
 
for a in A:
    for k in [4, 3, 2, 1, 0]:
        for v in range(1001 - a):
            dp[k+1][v+a] += dp[k][v]
print(dp[5][1000])

File: 3_3/p020.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

const (
	initBuffSize int = 10000
	maxBufSize int = 1000000
)

var scan = bufio.NewScanner(os.Stdin)

func init() {
	buf := make([]byte, initBuffSize)
	scan.Buffer(buf, maxBufSize)
}

func IntStdin() int {
	scan.Scan()
	num, err := strconv.Atoi(scan.Text())
	if err != nil {
		panic(err)
	}
	return num
}

func IntsStdin(n int) []int {
	nums := make([]int, n)

	scan.Scan()
	for i, v := range strings.Split(scan.Text(), " ") {
		num, err := strconv.Atoi(v)
		if err != nil {
			panic(err)
		}
		nums[i] = num
	}
	return nums
}

func main() {
	n := IntStdin()
	lst := IntsStdin(n)
	expected := 1000

	ans := 0
	for i := 0; i<n; i++ {
		for j := i+1; j<n; j++ {
			for k := j+1; k<n; k++ {
				for l := k+1; l<n; l++ {
					for m := l+1; m<n; m++ {
						if expected - (lst[i] + lst[j] + lst[k] + lst[l] + lst[m]) == 0 {
							ans += 1
						}
					}
				}
			}
		}
	}

	fmt.Println(ans)
}
